src/peergos/server/net/MultipartReceiver.java

```diff
@@ -0,0 +1,46 @@
+package peergos.server.net;
+
+import java.io.*;
+import java.util.*;
+
+public class MultipartReceiver {
+
+    public static byte[] extractFile(InputStream in, String boundary) {
+        try {
+            String first = readLine(in);
+            if (!first.substring(2).equals(boundary))
+                throw new IllegalStateException("Incorrect boundary! " + boundary + " != " + first.substring(2));
+            String filenameLine = readLine(in);
+            String contentType = readLine(in);
+            String encoding = contentType.length() > 0 ? readLine(in) : "";
+            String blank = encoding.length() > 0 ? readLine(in) : "";
+
+            ByteArrayOutputStream file = new ByteArrayOutputStream();
+            byte[] buffer = new byte[4096];
+            int r;
+            while ((r = in.read(buffer)) != -1)
+                file.write(buffer, 0, r);
+            byte[] bytes = file.toByteArray();
+            // 8 is two \r\n and two -- at end of stream
+            return Arrays.copyOfRange(bytes, 0, bytes.length - (8 + boundary.length()));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String readLine(InputStream in) throws IOException {
```

Comment:
Perhaps this  function should  take a  max-length parameter like the  Serialize.deserializeString method to stop reading an  infinitely long stream?